Assets/HotmaxEditorVR/loadPropsMenu.cs

```diff
@@ -0,0 +1,288 @@
+    }
+
+    void waitToLoadTriggerEvent()
+    {
+        inputManager.trackedController2.TriggerClicked += triggerClicked;
+    }
+
+
+    private void OnDisable()
+    {
+        inputManager.trackedController2.TriggerClicked -= triggerClicked;
+
+        stateManager.editorModeEvent -= updateEditorMode;
+    }
+
+    void updateEditorMode(stateManager.editorModes value)
+    {
+        _editorMode = value;
+
+        if (_editorMode == stateManager.editorModes.spawnMenuMode)
+        {
+            toggleMenuComponents(true);
+
+            //--make menu face the HMD
+            GameObject menu = transform.parent.transform.gameObject;
+            GameObject vrCam = init.vrCamera;
+
+            Vector3 forwardAmount = vrCam.transform.position + (vrCam.transform.forward * 2.15f);
+            Quaternion rotationAmount = vrCam.transform.rotation;
+
+            menu.transform.position = new Vector3(forwardAmount.x, vrCam.transform.position.y - 1, forwardAmount.z);
+
+            menu.transform.rotation = rotationAmount;
+            menu.transform.localEulerAngles = new Vector3(0, menu.transform.localEulerAngles.y + 180, 0);
+        }
+        else
+        {
+            toggleMenuComponents(false);
+        }
+    }
+
+    void Start()
+    {
+        _stateManagerMutatorRef = GameObject.FindObjectOfType(typeof(stateManager)) as stateManager;
+
+        propParent = GameObject.Find("props");
+
+        //get the names of all props in props
+        for (int i = 0; i < propParent.transform.childCount; i++)
+        {
+            propName.Add(propParent.transform.GetChild(i).gameObject.name);
+        }
+
+        //weed out props that have the same name
+        propName = propName.Distinct().ToList();
+
+        //copy the game objects with the unique names to a gameObject list 
+        for (int i = 0; i < propName.Count; i++)
+        {
+            propObject.Add(GameObject.Find(propName[i]));
+        }
+
+        //set all game objects in props inactive if they are not in the stage
+        if (!propIsInStageAlready)
+        {
+            for (int i = 0; i < propParent.transform.childCount; i++)
+            {
+                propParent.transform.GetChild(i).gameObject.SetActive(false);
+            }
+        }
+
+        //create "chunks" groups of 4 to display the object
+        chunkedList = ListExtensions.ChunkBy(propObject, 4);
+        showPropObjectPage(0);
+    }
+
+    void showPropObjectPage(int page)
+    {
+        currentPage = page;
+        //remove old tiles before loading new ones
+        for (int i = 0; i < propTiles.transform.childCount; i++)
+        {
+            Destroy(propTiles.transform.GetChild(i).gameObject);
+        }
+
+        //load and arrange "chunks", 4 objects from the propObject List 
+        for (int i = 0; i < chunkedList[page].Count; i++)
+        {
+            GameObject propTile = Instantiate(Resources.Load("propTile", typeof(GameObject))) as GameObject;
```

Comment:
what is the propTile resource? a plane?

Reply:
i might push this kind of information - like name, rotation, and gameobject in the constructor of a propTile class - that would on instantiation build up it's appearance, and also handle the iHittable event from raycaster

Reply:
or if you are using a prefab, you could just have that class on the "propTile" prefab, and push this info into the class's constructor directly from here.

Reply:
okay, got it

Reply:
I changed all of the menu buttons to iHittable interfaces